mongo_connector/test_utils.py

```diff
@@ -163,6 +163,21 @@ def get_config(self):
 
     def _init_from_response(self, response):
         self.id = response['id']
+        for _ in range(5):
+            states = set(member['state'] for member in response['members'])
+            if len(states) > 1:
+                if states.issuperset(set([1, 2])):
+                    # Make sure that the response contains a primary and
+                    # secondary.
+                    break
+            elif 1 in states:
+                # Make sure that the response contains a primary.
+                break
+            # Sometimes MO replies when the secondary is still in the STARTUP
+            # state. Wait a bit and retry.
```

Comment:
Thanks for the insight! Opened https://github.com/10gen/mongo-orchestration/pull/225